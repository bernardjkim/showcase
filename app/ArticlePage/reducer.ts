/*
 *
 * ArticlePage reducer
 *
 */

import { ArticleActionTypes, ArticleState, SET_ARTICLE_ID } from './types';

// The initial state of the App
export const initialState: ArticleState = {
  articleId: undefined,
};

function articlePageReducer(state: ArticleState = initialState, action: ArticleActionTypes) {
  switch (action.type) {
    case SET_ARTICLE_ID:
      return {
        ...state,
        articleId: action.articleId,
      };

    default:
      return state;
  }
}

export default articlePageReducer;
