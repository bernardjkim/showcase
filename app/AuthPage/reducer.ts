/*
 *
 * AuthPage reducer
 *
 */

// import { LOCATION_CHANGE } from 'react-router-redux';
import {
  AuthState,
  AuthActionTypes,
  CLEAR_ERRORS,
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
  CREATE_USER,
  CREATE_USER_ERROR,
  CREATE_USER_SUCCESS,
} from './types';

export const initialState: AuthState = {
  loading: false,
  error: undefined,
};

function AuthPageReducer(
  state: AuthState = initialState,
  action: AuthActionTypes,
) {
  switch (action.type) {
    // case LOCATION_CHANGE:
    //   return { ...state, error: undefined };

    case CLEAR_ERRORS:
      return { ...AbortController, error: undefined };

    case CREATE_TOKEN:
      return { ...state, loading: true, error: undefined };

    case CREATE_TOKEN_SUCCESS:
      return { ...state, loading: false };

    case CREATE_TOKEN_ERROR: {
      return { ...state, loading: false, error: { createToken: action.error } };
    }

    case CREATE_USER:
      return { ...state, loading: true, error: undefined };

    case CREATE_USER_SUCCESS:
      return { ...state, loading: false };

    case CREATE_USER_ERROR:
      return { ...state, loading: false, error: { createUser: action.error } };

    default:
      return state;
  }
}

export default AuthPageReducer;
