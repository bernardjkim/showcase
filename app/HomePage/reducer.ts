/*
 *
 * HomePage reducer
 *
 */

import {
  HomeState,
  HomeActionTypes,
  CLEAR_STATE,
  LOAD_ARTICLES_ALL,
  LOAD_ARTICLES_ALL_ERROR,
  LOAD_ARTICLES_ALL_SUCCESS,
  LOAD_NEXT,
  LOAD_NEXT_SUCCESS,
  LOAD_NEXT_ERROR,
} from './types';

export const initialState: HomeState = {
  loading: false,
  error: undefined,
  articles: [],
  offset: 0,
};

function homePageReducer(
  state: HomeState = initialState,
  action: HomeActionTypes,
) {
  switch (action.type) {
    case CLEAR_STATE:
      return initialState;

    case LOAD_NEXT:
      return {
        ...state,
        loading: true,
        error: undefined,
      };

    case LOAD_NEXT_SUCCESS:
      return {
        ...state,
        articles: state.articles.concat(action.articles),
        offset: state.offset + action.articles.length,
        loading: false,
      };

    case LOAD_NEXT_ERROR:
      return {
        ...state,
        error: action.error,
        loading: false,
      };

    case LOAD_ARTICLES_ALL:
      return {
        ...state,
        articles: [],
        loading: true,
        error: undefined,
      };

    case LOAD_ARTICLES_ALL_SUCCESS:
      return {
        ...state,
        articles: state.articles.concat(action.articles),
        offset: action.articles.length,
        loading: false,
      };

    case LOAD_ARTICLES_ALL_ERROR:
      return {
        ...state,
        error: action.error,
        loading: false,
      };

    default:
      return state;
  }
}

export default homePageReducer;
