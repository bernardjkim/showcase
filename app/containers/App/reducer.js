/*
 * 
 * App reducer
 * 
 */

import { fromJS } from 'immutable';
import {
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
} from './constants';

// The initial state of the App
export const initialState = fromJS({
  loading: false,
  error: false,
  token: false,
});

function appReducer(state = initialState, action) {
  switch (action.type) {
    case CREATE_TOKEN:
      return state.set('loading', true).set('error', false);

    case CREATE_TOKEN_SUCCESS:
      return state.set('token', action.token).set('loading', false);

    case CREATE_TOKEN_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    default:
      return state;
  }
}

export default appReducer;
