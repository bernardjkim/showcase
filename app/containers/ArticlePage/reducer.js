/*
 *
 * ArticlePage reducer
 *
 */

import { fromJS } from 'immutable';
import {
  CREATE_COMMENT,
  CREATE_COMMENT_ERROR,
  CREATE_COMMENT_SUCCESS,
  LIKE_ARTICLE,
  LIKE_ARTICLE_ERROR,
  LIKE_ARTICLE_SUCCESS,
  LOAD_ARTICLE,
  LOAD_ARTICLE_ERROR,
  LOAD_ARTICLE_SUCCESS,
  LOAD_COMMENTS,
  LOAD_COMMENTS_ERROR,
  LOAD_COMMENTS_SUCCESS,
} from './constants';

// The initial state of the App
export const initialState = fromJS({
  loading: false,
  error: false,
  /**
   * article
   * @field {string}  _id         - Article id
   * @field {string}  uri         - Article uri
   * @field {string}  github      - GitHub repo
   * @field {string}  description - Article description
   * @field {string}  image       - Article image link
   * @field {number}  likes       - Number of likes
   * @field {bool}    likedByUser - Article is liked by user
   * @field {array}   comments    - Array of comments
   */
  article: false,
});

function articlePageReducer(state = initialState, action) {
  switch (action.type) {
    case CREATE_COMMENT:
      return state.set('loading', true).set('error', false);

    case CREATE_COMMENT_SUCCESS:
      return state
        .updateIn(['article', 'comments'], comments =>
          comments.push(fromJS(action.comment)),
        )
        .set('loading', false);

    case CREATE_COMMENT_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    case LIKE_ARTICLE:
      return state.set('loading', true).set('error', false);

    case LIKE_ARTICLE_SUCCESS:
      return state
        .updateIn(['article', 'likes'], likes => likes + 1)
        .setIn(['article', 'likedByUser'], true)
        .set('loading', false);

    case LIKE_ARTICLE_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    case LOAD_ARTICLE:
      return state
        .set('loading', true)
        .set('error', false)
        .set('article', false);

    case LOAD_ARTICLE_SUCCESS:
      return state.set('article', fromJS(action.article)).set('loading', false);

    case LOAD_ARTICLE_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    case LOAD_COMMENTS:
      return state.set('loading', true).set('error', false);

    case LOAD_COMMENTS_SUCCESS:
      return state.setIn(['article', 'comments'], fromJS(action.comments));

    case LOAD_COMMENTS_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    default:
      return state;
  }
}

export default articlePageReducer;
