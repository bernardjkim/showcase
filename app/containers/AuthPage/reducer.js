/*
 *
 * AuthPage reducer
 *
 */

import { fromJS } from 'immutable';
import { LOCATION_CHANGE } from 'react-router-redux';
import {
  CLEAR_ERRORS,
  CREATE_TOKEN,
  CREATE_TOKEN_ERROR,
  CREATE_TOKEN_SUCCESS,
  CREATE_USER,
  CREATE_USER_ERROR,
  CREATE_USER_SUCCESS,
} from './constants';

export const initialState = fromJS({
  loading: false,
  error: false,
});

function AuthPageReducer(state = initialState, action) {
  switch (action.type) {
    case LOCATION_CHANGE:
      return state.set('error', false);

    case CLEAR_ERRORS:
      return state.set('error', false);

    case CREATE_TOKEN:
      return state.set('loading', true).set('error', false);

    case CREATE_TOKEN_SUCCESS:
      return state.set('loading', false);

    case CREATE_TOKEN_ERROR: {
      return state
        .set('error', fromJS({ createToken: action.error }))
        .set('loading', false);
    }

    case CREATE_USER:
      return state.set('loading', true).set('error', false);

    case CREATE_USER_SUCCESS:
      return state.set('loading', false);

    case CREATE_USER_ERROR:
      return state
        .set('error', fromJS({ createUser: action.error }))
        .set('loading', false);

    default:
      return state;
  }
}

export default AuthPageReducer;
