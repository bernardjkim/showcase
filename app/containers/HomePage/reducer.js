/*
 *
 * HomePage reducer
 *
 */

import { fromJS } from 'immutable';
import {
  LIKE_ARTICLE,
  LIKE_ARTICLE_ERROR,
  LIKE_ARTICLE_SUCCESS,
  LOAD_ARTICLE,
  LOAD_ARTICLE_ERROR,
  LOAD_ARTICLE_SUCCESS,
} from './constants';

// The initial state of the App
export const initialState = fromJS({
  loading: false,
  error: false,
  article: false,
});

function homePageReducer(state = initialState, action) {
  switch (action.type) {
    case LIKE_ARTICLE:
      return state.set('loading', true).set('error', false);

    case LIKE_ARTICLE_SUCCESS: {
      const article = state.get('article');
      article.likes += 1;
      return state.set('article', article).set('loading', false);
    }
    case LIKE_ARTICLE_ERROR:
      return state.set('error', action.error).set('loading', false);

    case LOAD_ARTICLE:
      return state
        .set('loading', true)
        .set('error', false)
        .set('article', false);

    case LOAD_ARTICLE_SUCCESS:
      return state.set('article', action.article).set('loading', false);

    case LOAD_ARTICLE_ERROR:
      return state.set('error', action.error).set('loading', false);

    default:
      return state;
  }
}

export default homePageReducer;
