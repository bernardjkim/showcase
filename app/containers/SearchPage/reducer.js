/*
 *
 * SearchPage reducer
 *
 */

import { fromJS } from 'immutable';
import {
  CLEAR_STATE,
  SET_SEARCH,
  LOAD_ARTICLES,
  LOAD_ARTICLES_ERROR,
  LOAD_ARTICLES_SUCCESS,
  LOAD_NEXT,
  LOAD_NEXT_ERROR,
  LOAD_NEXT_SUCCESS,
} from './constants';

export const initialState = fromJS({
  loading: false,
  error: false,
  articles: [],
  search: '',
  offset: 0,
});

function searchPageReducer(state = initialState, action) {
  switch (action.type) {
    case CLEAR_STATE:
      return initialState;

    case SET_SEARCH:
      return state.set('search', action.search);

    case LOAD_ARTICLES:
      return state
        .set('articles', fromJS([]))
        .set('offset', 0)
        .set('loading', true)
        .set('error', false);

    case LOAD_ARTICLES_SUCCESS:
      return state
        .set('articles', fromJS(action.articles))
        .set('offset', action.articles.length)
        .set('loading', false);

    case LOAD_ARTICLES_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    case LOAD_NEXT:
      return state.set('loading', true).set('error', false);

    case LOAD_NEXT_SUCCESS:
      return state
        .update('articles', articles =>
          articles.concat(fromJS(action.articles)),
        )
        .update('offset', val => val + action.articles.length)
        .set('loading', false);

    case LOAD_NEXT_ERROR:
      return state.set('error', fromJS(action.error)).set('loading', false);

    default:
      return state;
  }
}

export default searchPageReducer;
