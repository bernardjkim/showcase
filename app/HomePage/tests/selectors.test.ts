// import { fromJS } from 'immutable';
import {
  makeSelectArticles,
  makeSelectError,
  makeSelectLoading,
  selectHomePageDomain,
} from '../selectors';

import { initialState } from '../reducer';

describe('selectHomePageDomain', () => {
  it('should select the home page state', () => {
    expect(selectHomePageDomain(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectLoading', () => {
  const loadingSelector = makeSelectLoading();
  it('should select the loading state', () => {
    expect(loadingSelector(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectError', () => {
  const errorSelector = makeSelectError();
  it('should select the error state', () => {
    expect(errorSelector(initialState)).toMatchSnapshot();
  });
});

describe('makeSelectArticles', () => {
  const articlesSelector = makeSelectArticles();
  it('should select the artciles state', () => {
    expect(articlesSelector(initialState)).toMatchSnapshot();
  });
});
