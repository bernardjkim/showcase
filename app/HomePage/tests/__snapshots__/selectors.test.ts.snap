// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeSelectArticles should select the artciles state 1`] = `Array []`;

exports[`makeSelectError should select the error state 1`] = `undefined`;

exports[`makeSelectLoading should select the loading state 1`] = `false`;

exports[`selectHomePageDomain should select the home page state 1`] = `
Object {
  "articles": Array [],
  "error": undefined,
  "loading": false,
  "offset": 0,
  "search": "",
}
`;
